app/controllers/registrations_controller.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class RegistrationsController < Devise::RegistrationsController
+  def create # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
+    build_resource(sign_up_params)
+
+    resource.save
+    yield resource if block_given?
+    if resource.persisted?
+      ::ProfileCreatorService.call(resource)
```

Comment:
can you reasign resource or have profile creation after action?

```
after_action :create_profile, only: :create

.... 

private

def create_profile
  ::Profiles::Create.call(user: resource)
end
```
or reasign:
```
  def build_resource
    self.resource = call your service here
  end
```

Reply:
Not working